zilencer/models.py

```diff
@@ -135,7 +135,10 @@ class BaseRemoteCount(BaseCount):
     # The remote_id field is the id value of the corresponding *Count object
     # on the remote server.
     # It lets us deduplicate data from the remote server.
-    remote_id = models.IntegerField()
+    # Note: Some counts don't come from the remote server, but rather
+    # are stats we track on the bouncer server itself, pertaining to the remote server.
+    # E.g. mobile_pushes_requested::day. Such counts will set this field to None.
+    remote_id = models.IntegerField(null=True)
```

Comment:
Makes sense, added this as a separate commit.

> we probably want to make the zilencer code for validating RemoteRealmCount data sent by the self-hosted server not let them write rows with remote_id=null

We already ensure they can't submit data like that because we have validators on these dicts, with `check_int` on the `id` field:  https://github.com/zulip/zulip/blob/15c3abe03c37f657f73612b5c27436cc9407d32e/zilencer/views.py#L479-L492

i added an `AssertionError` in `validate_incoming_table_data` to be super sure - but it shouldn't be possible to begin with, so that's why `AssertionError` instead of some kind of normal error.

> (I guess we also need to add validation logic preventing this new CountStat from being uploaded in the commit adding the new CountStat).

That's just the last commit in this series, right? We can squash it with its preceding commit (which add the CountStat) if we think it's important to have it like that - but it's probably not that sensitive and keeping them separate may be better for readability's sake? This doesn't seem like a valuable enough attack angle to hunt for a gap between us deploying those two commits.